### PR TITLE
Add class to serve as model for nosepass json report

### PR DESCRIPTION
**Pull request by julianoborba**

### CODE REVIEW

File: nosepass/report.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        self.tor = tor
+        self.waf = waf
+        self.open_ports = open_ports
+        self.alerts = alerts
+        self.score = score
+
+    def get_report_as_json(self, unpicklable):
+        return encode(self, unpicklable)
+
+    @staticmethod
```
Comment (resilva87) [inline nosepass/report.py]: @julianoborba kind of curious: does this method need to be static?

File: tests/test_report.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+                '"Content-security-policy" not present!',
+                '"Path" defined as "/"!'
+            ],
+            8
+        )
+
+        with open(f'{TESTS_PATH}/resources/report.json') as json_file:
+            cls.json = json_file.read()
+
+    def test_that_should_serialize_report_to_json(self):
```
Comment (resilva87) [inline tests/test_report.py]: @julianoborba I think you should test when both marshalling and unmarshalling the report fails

File: nosepass/report.py (hunk 1 above)
Comment (julianoborba) [reply nosepass/report.py]: Pycharm say "yes, is needed", but we should trust him?

File: tests/test_report.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        json = self.report_schema.dumps(self.report)
+
+        self.assertEqual(loads(self.json), loads(json))
+
+    def test_that_should_not_serialize_missing_fields_to_json(self):
+        json = self.report_schema.dumps(self.bad_report)
+
+        self.assertNotEqual(loads(self.json), loads(json))
+        self.assertTrue('waf' not in json)
```
Comment (carloosamarante) [inline tests/test_report.py]: What do you think about https://docs.python.org/3.7/library/unittest.html#unittest.TestCase.assertNotIn ? i think it's nice for this example =)

Comment (julianoborba) [reply tests/test_report.py]: #pride

**Pull request merged**